contracts/facets/securesecoMonetization/SearchSECOMonetizationFacet.sol

```diff
@@ -38,18 +40,25 @@ contract SearchSECOMonetizationFacet is AuthConsumer {
     function payForHashes(uint _amount, string memory _uniqueId) external {
         LibSearchSECOMonetizationStorage.Storage
             storage s = LibSearchSECOMonetizationStorage.getStorage();
-        ERC20 tokenContract = ERC20(address(this));
+        IERC20 tokenContract = IERC20(address(this));
 
         // Require that the balance of the sender has sufficient funds for this transaction
         // hashCost is the cost of a single hash
-        require(tokenContract.balanceOf(msg.sender) > s.hashCost * _amount, "Insufficient funds for this transaction");
+        require(tokenContract.balanceOf(msg.sender) > s.hashCost * _amount, "Insufficient tokens for this transaction");
 
-        tokenContract.transferFrom(msg.sender, address(this), s.hashCost * _amount);
+        // tokenContract.transferFrom(msg.sender, address(this), s.hashCost * _amount);
+        IBurnableGovernanceStructure burnBridges = IBurnableGovernanceStructure(address(this));
+        burnBridges.burnVotingPower(msg.sender, s.hashCost * _amount);
         
         // Emit event so back-end can verify payment
         emit PaymentProcessed(msg.sender, _amount, _uniqueId);
     }
 
+    function giveMoneyz() external {
```

Comment:
Make name more professional

Reply:
Sorry thanks for noticing! Updated it to mintTokensToSender